lib/services/awsAdapter.js

```diff
@@ -261,6 +272,93 @@ function AWSAdapter (options) {
 		}
 	};
 
+	this.getKeyPair = function (createKeyName, existingKeyName) {
+
+		var keyPairParams;
+
+		var keyPair = {
+			keyLocation : undefined,
+			name        : undefined
+		};
+
+		if (createKeyName === undefined) {
+			if (existingKeyName === undefined) {
+				//Default name and location of the Service Maker key-pair.
+				existingKeyName = "service-maker";
+				keyPair.keyLocation = DEFAULT_BUCKET + existingKeyName + ".pem";
+			}
+
+			keyPair.name = existingKeyName;
+
+			keyPairParams = {
+				KeyNames : [ existingKeyName ]
+			};
+
+			return ec2.describeKeyPairsAsync(keyPairParams)
+			.then(function () {
+				return keyPair;
```

Comment:
The thought behind this was that service maker wouldn't know where the private key would be. A user could use a key that only they have.